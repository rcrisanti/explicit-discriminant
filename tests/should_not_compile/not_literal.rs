use explicit_discriminant::ExplicitDiscriminant;

#[derive(ExplicitDiscriminant)]
#[pattern(1)]
enum Test {
    Zero = 0,
    One = 1,
}

fn main() {
    let t = Test::One;
}
