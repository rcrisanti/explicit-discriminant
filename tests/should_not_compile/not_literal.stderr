error: discriminant does not match any pattern
 --> tests/should_not_compile/not_literal.rs:6:12
  |
6 |     Zero = 0,
  |            ^
